File: backend/src/main/java/com/plantcare/plantcareassistant/dto/UserPlantDto.java
package com.plantcare.plantcareassistant.dto;

import com.plantcare.plantcareassistant.entities.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserPlantDto {
    private String apiPlantId;
    private String scientificName;
    private String customName;
    private String pictureUrl;

    private Boolean notificationsEnabled;
    private LocalTime notificationTime;
    private int wateringFrequency; // Frequency in days
    private String frequencyUnit; // "days", "weeks", "months"

}

File: AndroidApp/app/src/main/java/com/example/plantcare/data/model/UserPlantDto.kt
package com.example.plantcare.data.model

import java.time.LocalTime

data class UserPlantDto(
    val apiPlantId: String,
    val scientificName: String?,
    val customName: String?,
    val pictureUrl: String?,
    val notificationsEnabled: Boolean?,
    val notificationTime: LocalTime?,
    val wateringFrequency: Int, // Frequency in days
    val frequencyUnit: String // "days", "weeks", "months"
)

